### PR TITLE
11.0 mig mass editing

### PR DESCRIPTION
**Pull request by jnegrete**

### CODE REVIEW

File: mass_editing/ChangeLog.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
+===============================================================================
```
Comment (yajo) [inline mass_editing/ChangeLog.txt]: I vote to delete this file. Hard to maintain, and we have git for this.

File: mass_editing/README.rst [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+* Go to *Settings / Mass Editing / Mass Editing* and configure the object and fields for Mass Editing.
+
+Usage
+=====
+
+This module allows to add, update or remove the values of more than one records on the fly at the same time.
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/149/9.0
```
Comment (yajo) [inline mass_editing/README.rst]: 11.0

File: mass_editing/README.rst [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+* This way you can set / remove the values of the fields.
+
+.. image:: /mass_editing/static/description/mass_editing-5.png
+   :width: 70%
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/OCA/server-tools/issues>`_. In case of trouble, please
```
Comment (yajo) [inline mass_editing/README.rst]: server-ux

File: mass_editing/README.rst [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+check there if your issue has already been reported. If you spotted it first,
+help us smashing it by providing a detailed and welcomed `feedback
+<https://github.com/OCA/
+server-tools/issues/new?body=module:%20
+server-tools%0Aversion:%20
+9.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (yajo) [inline mass_editing/README.rst]: update template

File: mass_editing/README.rst [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+   :alt: Odoo Community Association
+   :target: http://odoo-community.org
+
+This module is maintained by the OCA.
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit http://odoo-community.org.
```
Comment (yajo) [inline mass_editing/README.rst]: https

File: mass_editing/README.rst [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+* Jay Vora <jay.vora@serpentcs.com>
+* Jairo Llopis <jairo.llopis@tecnativa.com>
+* Juan Negrete <jnegrete@casasalce.com>
+
+Maintainer
+----------
+
+.. image:: https://odoo-community.org/logo.png
+   :alt: Odoo Community Association
+   :target: http://odoo-community.org
```
Comment (yajo) [inline mass_editing/README.rst]: https

File: mass_editing/README.rst [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+This module is maintained by the OCA.
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit http://odoo-community.org.
+
```
Comment (yajo) [inline mass_editing/README.rst]: remove

File: mass_editing/__manifest__.py [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    'name': 'Mass Editing',
+    'version': '11.0.1.0.0',
+    'author': 'Serpent Consulting Services Pvt. Ltd., '
+              'Tecnativa, '
+              'Odoo Community Association (OCA)',
+    'category': 'Tools',
+    'website': 'http://www.serpentcs.com',
+    'license': 'GPL-3 or any later version',
```
Comment (yajo) [inline mass_editing/__manifest__.py]: AGPL-3

File: mass_editing/i18n/mass_editing.pot [hunk 9]
```diff
@@ -0,0 +1,129 @@
+# Translation of OpenERP Server.
+# This file contains the translation of the following modules:
+#	* mass_editing
```
Comment (yajo) [inline mass_editing/i18n/mass_editing.pot]: delete this file

File: mass_editing/models/mass_object.py [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+                                           'Sidebar action',
+                                           readonly=True,
+                                           help="Sidebar action to make this "
+                                                "template available on "
+                                                "records of the related "
+                                                "document model.")
+    model_list = fields.Char('Model List')
+
+    _sql_constraints = [
+        ('name_uniq', 'unique (name)', _('Name must be unique!')),
```
Comment (yajo) [inline mass_editing/models/mass_object.py]: Remove `_()`

File: mass_editing/models/mass_object.py [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+            'binding_type': 'action',
+            'multi': True,
+        }).id
+        self.write(vals)
+        return True
+
+    @api.multi
+    def unlink_action(self):
+        # We make sudo as any user with rights in this model should be able
+        # to delete the action, not only admin
```
Comment (yajo) [inline mass_editing/models/mass_object.py]: This should be achieved by a `ir.rule` instead.

File: mass_editing/README.rst [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+Mass Editing
+============
+
+This module provides the following features:
+
+* You can add, update or remove the values of more than one records on the fly at the same time.
+
+* You can configure mass editing for any Odoo model.
+
+* The video explaining the features and how-to for OpenERP Version 6 is here http://t.co/wukYMx1A
```
Comment (yajo) [inline mass_editing/README.rst]: 🔥

File: mass_editing/README.rst [hunk 13]
```diff
@@ -0,0 +1,112 @@
...
+
+* The video explaining the features and how-to for OpenERP Version 7 is here : http://www.youtube.com/watch?v=9BH0o74A748&feature=youtu.be
+
+* For more details/customization/feedback contact us on contact@serpentcs.com
```
Comment (yajo) [inline mass_editing/README.rst]: 🔥

File: mass_editing/README.rst [hunk 14]
```diff
@@ -0,0 +1,112 @@
+.. image:: https://img.shields.io/badge/license-LGPLv3-blue.svg
+   :target: https://www.gnu.org/licenses/lgpl.html
+   :alt: License: LGPL-3
```
Comment (yajo) [inline mass_editing/README.rst]: AGPL

**Review by yajo — CHANGES_REQUESTED** [review]

**Comment by yajo** [comment]
Please could you also check that this migration is not affected by https://github.com/OCA/server-tools/issues/959?

**Comment by RoelAdriaans** [comment]
```
Fout:
TypeError: val is null

http://localhost/web/static/src/js/views/basic/basic_model.js:3325
Traceback:
_parseServerData/<@http://localhost/web/static/src/js/views/basic/basic_model.js:3325:29
_.forEach@http://localhost/web/static/lib/underscore/underscore.js:145:9
_parseServerData@http://localhost/web/static/src/js/views/basic/basic_model.js:3311:9
_fetchRecord/<@http://localhost/web/static/src/js/views/basic/basic_model.js:1838:17
then/</</<@http://localhost/web/static/lib/jquery/jquery.js:3276:86
fire@http://localhost/web/static/lib/jquery/jquery.js:3119:38
fireWith@http://localhost/web/static/lib/jquery/jquery.js:3231:49
then/</</<@http://localhost/web/static/lib/jquery/jquery.js:3283:73
fire@http://localhost/web/static/lib/jquery/jquery.js:3119:38
fireWith@http://localhost/web/static/lib/jquery/jquery.js:3231:49
Deferred/</deferred[tuple[0]]@http://localhost/web/static/lib/jquery/jquery.js:3321:33
rpc/</<@http://localhost/web/static/src/js/services/ajax_service.js:13:21
then/</</<@http://localhost/web/static/lib/jquery/jquery.js:3276:86
fire@http://localhost/web/static/lib/jquery/jquery.js:3119:38
fireWith@http://localhost/web/static/lib/jquery/jquery.js:3231:49
Deferred/</deferred[tuple[0]]@http://localhost/web/static/lib/jquery/jquery.js:3321:33
fire@http://localhost/web/static/lib/jquery/jquery.js:3119:38
fireWith@http://localhost/web/static/lib/jquery/jquery.js:3231:49
then/</</<@http://localhost/web/static/lib/jquery/jquery.js:3283:73
fire@http://localhost/web/static/lib/jquery/jquery.js:3119:38
fireWith@http://localhost/web/static/lib/jquery/jquery.js:3231:49
then/</</<@http://localhost/web/static/lib/jquery/jquery.js:3283:73
fire@http://localhost/web/static/lib/jquery/jquery.js:3119:38
fireWith@http://localhost/web/static/lib/jquery/jquery.js:3231:49
done@http://localhost/web/static/lib/jquery/jquery.js:9275:33
callback@http://localhost/web/static/lib/jquery/jquery.js:9685:57
```

Same error as @CasVissers has in https://github.com/OCA/server-tools/pull/1048#issuecomment-344580895
This dump is with debug=assets .

**Comment by jnegrete** [comment]
Could you explain me how can I reproduce this error?

**Comment by RoelAdriaans** [comment]
@jnegrete Just by using the module.. 
It seems to work for simple fields, but not for many2one, like the country.
See this video: https://youtu.be/I_2aAcgzyf8

**Comment by astirpe** [comment]
I got this while searching a field:

```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/opt/odoo/odoo-11.0/odoo/http.py", line 646, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/opt/odoo/odoo-11.0/odoo/http.py", line 307, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/opt/odoo/odoo-11.0/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/opt/odoo/odoo-11.0/odoo/http.py", line 683, in dispatch
    result = self._call_function(**self.params)
  File "/opt/odoo/odoo-11.0/odoo/http.py", line 339, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/opt/odoo/odoo-11.0/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/opt/odoo/odoo-11.0/odoo/http.py", line 332, in checked_call
    result = self.endpoint(*a, **kw)
  File "/opt/odoo/odoo-11.0/odoo/http.py", line 927, in __call__
    return self.method(*args, **kw)
  File "/opt/odoo/odoo-11.0/odoo/http.py", line 512, in response_wrap
    response = f(*args, **kw)
  File "/opt/odoo/odoo-server/addons/web/controllers/main.py", line 866, in search_read
    return self.do_search_read(model, fields, offset, limit, domain, sort)
  File "/opt/odoo/odoo-server/addons/web/controllers/main.py", line 888, in do_search_read
    offset=offset or 0, limit=limit or False, order=sort or False)
  File "/opt/odoo/odoo-11.0/odoo/models.py", line 4118, in search_read
    records = self.search(domain or [], offset=offset, limit=limit, order=order)
  File "/opt/odoo/custom/addons/mass_editing/models/ir_model_fields.py", line 18, in search
    list(map(int, domain[2][1:-1].split(','))))]
ValueError: invalid literal for int() with base 10: 'nvo'
```

**Comment by jnegrete** [comment]
@Yajo could you tell me if I forgot something about the requested changes?
I'm ready to squash my commits, what do you think?

**Review by yajo — APPROVED** [review]
I think you can proceed with the squash.

All my remarks were solved, although they were not much more than stylistic... I think somebody more experienced in this addon should review too, because I see some logical changes that I don't understand...

**Comment by astirpe** [comment]
The error I got seems not reproducible anymore after the last commit. But I didn't test the module yet.

**Comment by ddufresne** [comment]
:+1: functional review

**Comment by pedrobaeza** [comment]
Please squash all adjacent translation commits together, and you can also merge some commits by author/logic group. After that, I will merge. Thanks for the work!

File: mass_editing/i18n/am.po [hunk 15]
```diff
@@ -0,0 +1,187 @@
+# Translation of Odoo Server.
+# This file contains the translation of the following modules:
+# * mass_editing
+# 
+# Translators:
+# OCA Transbot <transbot@odoo-community.org>, 2016
+msgid ""
+msgstr ""
+"Project-Id-Version: Odoo Server 9.0c\n"
```
Comment (JayVora-SerpentCS) [inline mass_editing/i18n/am.po:9]: Still has the old translation?

Comment (pedrobaeza) [reply mass_editing/i18n/am.po:9]: I think it's because headers are not updated unless you export from Odoo.

**Comment by pedrobaeza** [comment]
I have made myself the squash, so I finally merge this.

**Comment by rafaelbn** [comment]
There are 3 PRs related with this module opened in previous version, maybe same issue are in v11 if they are not taken in care

- https://github.com/OCA/server-tools/pull/848
- https://github.com/OCA/server-tools/pull/849 
- https://github.com/OCA/server-tools/pull/786

Just to point them

Regards

**Pull request merged**